synapse/http/client.py

```diff
@@ -819,6 +820,125 @@ def __init__(
             )
 
 
+class SimpleReplicationClient(BaseHttpClient):
```

Comment:
Can we rename this to just `ReplicationClient`? I find using something like "Simple" is...belittling when it isn't simple. 😄

Reply:
I'm ok with that. OTOH, is `SimpleHttpClient` simple with it's wrapping agent, ip-blocking and proxy support? 🤨

Reply:
Fixed in [c0096a3](https://github.com/matrix-org/synapse/pull/15470/commits/c0096a3e3cb05bd5c0504da697d164fdc2538b36)